### PR TITLE
Added Temperature pybind11 interface

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

## Summary

This PR creates a Temperature Pybind11 interface

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: include/ignition/math/Temperature.hh [hunk 1]
```diff
@@ -151,6 +151,16 @@ namespace ignition
       /// \return Reference to this instance
       public: Temperature &operator=(const Temperature &_temp);
 
+      /// \brief Addition operator
+      /// \param[in] _temp Temperature in Kelvin
+      /// \return Resulting temperature
+      public: Temperature operator+(const double _temp) const;
```
Comment (chapulina) [inline include/ignition/math/Temperature.hh:157]: We need to add tests for the new functions so the coverage doesn't drop

**Review by chapulina — CHANGES_REQUESTED** [review]

File: src/Temperature.cc [hunk 2]
```diff
@@ -160,6 +160,18 @@ Temperature Temperature::operator+(const Temperature &_temp)
...
+/////////////////////////////////////////////////
+Temperature Temperature::operator+(const double _temp) const
+{
+  return this->dataPtr->kelvin + _temp;
+}
+
+/////////////////////////////////////////////////
+Temperature Temperature::operator+(const Temperature &_temp) const
+{
+  return this->dataPtr->kelvin + _temp;
```
Comment (chapulina) [inline src/Temperature.cc:172]:
```suggestion
  return this->dataPtr->kelvin + _temp.Kelvin();
```

I think this may be needed

**Review by chapulina — APPROVED** [review]
LGTM with happy CI

**Comment by osrf-triage** [comment]
This pull request has been mentioned on **Gazebo Community**. There might be relevant details there:

https://community.gazebosim.org/t/new-ignition-releases-2022-03-01-citadel-edifice-fortress/1313/1

**Pull request merged**